### PR TITLE
Merging `aux-eff` to master

### PR DESCRIPTION
**Pull request by Huy-Ngo**
I have written several parts of `alure.pxd` and `palace.pyx` that are related to the class `AuxiliaryEffectSlot`. Please review and tell me if there are any mistakes.

### CODE REVIEW

**Comment by Huy-Ngo** [comment]
I suppose I have to resolve conflicts with `master` first then?

**Comment by McSinyx** [comment]
Now I believe it's unwise to let you mess around with git on a colab repo (I git gud with git after bunch of personal project where I could fuck up as hard as I want), yep, just resolve conflict on GitHub then.  the `@@` typo makes me believe you haven't test build at all.  Please hold the conflict resolve for my next commit, then you can run at least linting and building tests.

**Comment by McSinyx** [comment]
Ugh you can resolve the conflicts now.  Probably we need to squash everything together at merge to avoid noise.

> Please hold the conflict resolve for my next commit, then you can run at least linting and building tests.

File: palace.pyx [hunk 1]
```diff
@@ -1160,6 +1162,100 @@ cdef class SourceGroup:
...
+        """Retrieve each Source object and its pairing send this effect slot is
+        equivalent to calling `getSourceSends().size()`.
+        """
+        return self.impl.destroy()
+
+    @property
+    def source_sends(self) -> vector[SourceSend]:
+        return self.impl.get_source_sends()
+
+    @@property
```
Comment (9a24f0) [inline palace.pyx]: typo here

File: palace.pyx [hunk 2]
```diff
@@ -1160,6 +1162,100 @@ cdef class SourceGroup:
         """
         self.impl.destroy()
 
+
+cdef class AuxiliaryEffectSlot:
+    """An effect processor.
+
+    It takes the output mix of zero or more sources, applies DSP for the desired effect
```
Comment (9a24f0) [inline palace.pyx:1169]: You should specifies what is "it" before using, in alure#17 specified that is an effect processor.

Comment (McSinyx) [reply palace.pyx:1169]: that's 2 lines above.  Could you please proof read the whole thing, run the test (e.g. `tox -e py37`) and give a full review?

Comment (9a24f0) [reply palace.pyx:1169]: lemme finish my first commit so I can check his branch without warnings about being overwritten

Comment (McSinyx) [reply palace.pyx:1169]: There're problems all over the place.  You should be able to spot them.  Please give a proper review.

**Comment by McSinyx** [comment]
This is to much of a WIP and I'm needing to rebase master, so I'm closing the PR now.  After I finished rebasing, fix the patch by
```sh
git checkout master
git branch -D aux-eff
git checkout -b aux-eff
git diff 3e4e413a07f6d5e498faf520a469c845a1686b5c 990e2b8060d0b20c6c8402cfbae525d16dd17950 | git apply
git commit -am "Start working on AuxiliaryEffectSlot"
git push --set-upstream origin aux-eff --force-with-lease
```
and re-open the PR.

**Comment by McSinyx** [comment]
Aight auto build and test are working, you can rebase the patch on master now.

**Comment by Huy-Ngo** [comment]
Reopen #19

**Pull request closed without merging**